contracts/nft/contracts/solana/tests/universal-nft-cycle.ts

```
@@ -0,0 +1,1398 @@
+        });
+
+        it("Should implement replay protection with origin tracking", async () => {
+            console.log("🛡️  Test 5.3: Replay Protection with Origin Tracking");
+
+            // Get current nonce
+            const collectionAccount = await program.account.collection.fetch(collectionPda);
+            const currentNonce = collectionAccount.nonce.toNumber();
+
+            console.log(`      📊 Current nonce: ${currentNonce}`);
+            // Verify origin system contributes to replay protection
+            const collectionAccount = await program.account.collection.fetch(collectionPda);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)